src/core/connected_clients.rs

```diff
@@ -11,16 +11,53 @@ use bevy::{
 use crate::core::{replicon_tick::RepliconTick, ClientId};
 use client_visibility::ClientVisibility;
 
-/// Stores information about connected clients.
+#[derive(Resource, Default)]
+pub struct ConnectedClients {
```

Comment:
I don't entirely understand what you want me to do here. Keep `ConnectedClients` under core, move `ReplicatedClient(s)` to `server::replicated_clients`, and move `client_visibility` to `server::replicated_clients::client_visibility`?

Reply:
Correct!
And use `ConnectedClients` instead of `ReplicatedClients` for events for `SendMode::Broadcast` and `SendMode::BroadcastExcept` instead of `ReplicatedClients`.

Reply:
There is a problem with having events be sent out using ConnectedClients:
```rs
unsafe fn send_with<E: Event>(
    event_data: &ServerEvent,
    ctx: &mut ServerSendCtx,
    event: &E,
    mode: &SendMode,
    server: &mut RepliconServer,
    connected_clients: &ConnectedClients,
) -> bincode::Result<()> {
    match *mode {
        SendMode::Broadcast => {
            let mut previous_message = None;
            for client in connected_clients.iter() {
                let message = serialize_with(event_data, ctx, event, client, /* <--- HERE */ previous_message)?;
                server.send(client.id() /* <-- HERE */, event_data.channel_id, message.bytes.clone());
```
`serialize_with` takes a `ReplicatedClient`, which we aren't guaranteed to have since we're only working with connected clients who may not have replication yet. It uses the ReplicatedClient only for `client.init_tick()`, in:
```rs
unsafe fn serialize_with<E: Event>(
    event_data: &ServerEvent,
    ctx: &mut ServerSendCtx,
    event: &E,
    client: &ReplicatedClient,
    previous_message: Option<SerializedMessage>,
) -> bincode::Result<SerializedMessage> {
```
The `client.id()` can just be replaced with `client`, since that already *is* the client ID.

Does this mean we need to track init tick separately from replicated client state? Is there something I'm missing?

Reply:
Ah, my bad, I forgot about at `serialize_with`. Then discard this suggestion.

But I would move `ConnectedClients` into a separate module under `core`. Just a personal preference :)

Reply:
Yep already moved ConnectedClients to core under my next commit. But you still raise a good point about sending out events to all *connected* clients, not necessarily replicated ones. Since otherwise, how will non-replicated clients receive any events? The entire point of this PR is to allow disabling replication on connection, *but still allow sending/receiving events*. So this might actually be an implementation issue that I've overlooked.

Reply:
Right, you won't be able to send events...
This is why I suggested to add a test for events. Tests sometimes help to discover such issues.
Let me think about it a bit more and I will back to you. Consider this suggestion as resolved.